src/Blocks/Button.purs

```diff
@@ -2,70 +2,336 @@ module Ocelot.Block.Button where
 
 import Prelude
 
-import DOM.HTML.Indexed (HTMLbutton)
+import DOM.HTML.Indexed (HTMLbutton, HTMLdiv)
 import Halogen.HTML as HH
 import Halogen.HTML.Properties as HP
 import Ocelot.Core.Utils ((<&>))
 
-data ButtonType
-  = Default
-  | Primary
-  | Secondary
-  | Transparent
+buttonSharedClasses :: Array HH.ClassName
+buttonSharedClasses = HH.ClassName <$>
+  [ "no-outline"
+  , "px-4"
+  , "py-3"
+  , "!active:border-b"
+  , "active:border-t"
+  , "disabled:opacity-50"
+  , "disabled:cursor-default"
+  ]
 
-type ButtonProps =
-  { type_ :: ButtonType }
+buttonMainClasses :: Array HH.ClassName
+buttonMainClasses = buttonSharedClasses <>
+  ( HH.ClassName <$>
+    [ "rounded"
+    ]
+  )
 
 buttonClasses :: Array HH.ClassName
 buttonClasses = HH.ClassName <$>
-  [ "no-outline"
-  , "px-4"
-  , "py-3"
-  , "rounded"
-  , "shadow"
+  [ "bg-grey-50-a20"
+  , "border-grey-50-a20"
+  , "hover:!disabled:bg-grey-50-a30"
+  , "focus:bg-grey-50-a30"
+  , "text-black-20"
   ]
 
-buttonTypeClasses :: ButtonType -> Array HH.ClassName
-buttonTypeClasses type_ = HH.ClassName <$> classNames type_
-  where
-    classNames Default =
-      [ "bg-grey-50-a20"
-      , "hover:bg-grey-50-a30"
-      , "text-black-20"
-      ]
-
-    classNames Primary =
-      [ "bg-blue-88"
-      , "hover:bg-blue-82"
-      , "text-white"
-      ]
-
-    classNames Secondary =
-      [ "bg-grey-70-a30"
-      , "hover:bg-grey-70-a40"
-      , "text-grey-70"
-      ]
-
-    classNames Transparent =
-      [ "bg-transparent"
-      , "text-grey-70"
-      ]
+buttonPrimaryClasses :: Array HH.ClassName
+buttonPrimaryClasses = HH.ClassName <$>
+  [ "bg-blue-88"
+  , "border-blue-88"
+  , "hover:!disabled:bg-blue-82"
+  , "focus:bg-blue-82"
+  , "text-white"
+  ]
 
-button_
+buttonDarkClasses :: Array HH.ClassName
+buttonDarkClasses = HH.ClassName <$>
+  [ "bg-grey-70-a30"
+  , "border-grey-70-a30"
+  , "hover:!disabled:bg-grey-70-a40"
+  , "focus:bg-grey-70-a40"
+  , "text-grey-70"
+  ]
+
+buttonClearClasses :: Array HH.ClassName
+buttonClearClasses = HH.ClassName <$>
+  [ "bg-transparent"
+  , "border-transparent"
+  , "text-grey-70"
+  , "hover:text-grey-70-a30"
+  , "focus:text-grey-70-a30"
+  ]
+
+buttonGroupClasses :: Array HH.ClassName
+buttonGroupClasses = HH.ClassName <$>
+  [ "flex"
+  , "items-center"
+  ]
+
+centerClasses :: Array HH.ClassName
+centerClasses = HH.ClassName <$>
+  [ "mr-px"
+  ]
+
+leftClasses :: Array HH.ClassName
+leftClasses = HH.ClassName <$>
+  [ "mr-px"
+  , "rounded-l"
+  ]
+
+rightClasses :: Array HH.ClassName
+rightClasses = HH.ClassName <$>
+  [ "rounded-r"
+  ]
+
+buttonBuilder
   :: ∀ p i
-   . ButtonProps
+   . Array HH.ClassName
+  -> Array (HH.IProp HTMLbutton i)
   -> Array (HH.HTML p i)
   -> HH.HTML p i
-button_ { type_ } =
+buttonBuilder classes iprops =
   HH.button
-    [ HP.classes (buttonTypeClasses type_ <> buttonClasses) ]
+    ( [ HP.classes $ buttonMainClasses <> classes ] <&> iprops )
 
 button
```

Comment:
Switch from `Button.button` and `Button.buttonPrimary` to `Button.default` and `Button.primary`